github/client.go

```diff
@@ -247,6 +247,27 @@ func (client *Client) CreateRepository(project *Project, description, homepage s
 	return
 }
 
+func (client *Client) DeleteRepository(project *Project) (err error) {
+	var repoURL string
+	if project.Owner != client.Host.User {
+		repoURL = fmt.Sprintf("/repos/%s/%s", project.Owner, project.Name)
+	} else {
+		repoURL = fmt.Sprintf("/repos/%s/%s", client.Host.User, project.Name)
+	}
+
+	api, err := client.simpleApi()
+	if err != nil {
+		return
+	}
+
+	res, err := api.Delete(repoURL)
+	if err = checkStatus(204, "deleting repository", res, err); err != nil {
```

Comment:
The `if err != nil { return err }` conditional isn't necessary since the function already does `return err` in the end.